visualizer/src/service/tools/segment/toolUtils.js

```diff
@@ -3,16 +3,16 @@ import { assign } from 'xstate';
 // actions to save data from events in context
 export const toolActions = {
   setCoordinates: assign((_, { x, y }) => ({ x, y })),
-  setLabel: assign((_, { label }) => ({ label })),
-  setForeground: assign({ foreground: (_, { foreground }) => foreground }),
-  setBackground: assign({ background: (_, { background }) => background }),
-  setSelected: assign({ selected: (_, { selected }) => selected }),
+  setHovering: assign((_, { hovering }) => ({ hovering })),
+  setForeground: assign((_, { foreground }) => ({ foreground })),
+  setBackground: assign((_, { background }) => ({ background })),
+  setSelected: assign((_, { selected }) => ({ selected })),
```

Comment:
This was to make each line consistent. There are two ways assign actions can be written
```assign({ foreground: (_, ({ foreground }) => foreground })```
or 
```assign((_, { foreground }) => ({ foreground }))```
Here I changed all of them to the later because it was mixed before